src/libraries/System.Private.CoreLib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -188,11 +189,12 @@ private protected sealed override unsafe int GetByteCountFast(char* pChars, int
 
             if (!(fallback is EncoderReplacementFallback replacementFallback
                 && replacementFallback.MaxCharCount == 1
-                && replacementFallback.DefaultString[0] <= 0x7F))
+                && Ascii.IsValid(replacementFallback.DefaultString[0])))
```

Comment:
`Ascii.IsValid(char)` was added mostly for completeness and having everything in once place.

> Should we have Array.IsValid(int value)

I assume you meant `Ascii.IsValid(int)`? On the one hand it would be more flexible (users could pass integers as inputs too), on the other I am not sure about codegen differences and whether everyone knows that they can pass byte/char to int-accepting method.

Reply:
> Ascii.IsValid(char) was added mostly for completeness and having everything in once place.

So which should we be using?  Should we change all use of char.IsAscii to be Ascii.IsValid?

Reply:
Please no :) Personally I find `char.IsAscii` to be more readable.

Reply:
Then should we replace all Ascii.IsValid(char) with char.IsAscii?

We "just" added char.IsAscii in .NET 6.  Now in .NET 8 we're adding an identical Ascii.IsValid(char).  I'm trying to rationalize why we have both.  I get the consistency argument, but I don't think it's worth consistency just to have a method no one uses.

Reply:
@adamsitnik, opinions?